MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -1107,10 +1107,16 @@ public PlayAccess GetPlayAccess(User user)
         }
 
         public virtual List<MediaStream> GetMediaStreams()
```

Comment:
Instead of adding a new method I considered changing this to have an optional, null-default `MediaStreamType?` parameter. That requires changing the `IHasMediaSources` signature to include the optional parameter. Is that allowed/desired?

There are places where `GetMediaStreams` needs to return all streams but also several places that immediately filter the list of streams to a specific type so the method will be called with and without the type argument.

Reply:
It would be better to simply use the MediaSourceManager directly instead of via BaseItem

Reply:
I was initially reluctant to do this when I remembered that `LiveTvChannel` overrides `GetMediaStreams` to return an empty list (and I had missed the corresponding override for my changes), but none of the places that I switched to the new call can be reached with that type anyway. Converting this to a draft until I finish switching over.